### PR TITLE
IoTConsensusV2: Fix syncLag error when using steam mode and 2+ replicas

### PR DESCRIPTION
**Pull request by Pengzna**
as title.

### CODE REVIEW

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/pipe/metric/PipeConsensusSyncLagManager.java [hunk 1]
```diff
@@ -33,66 +34,50 @@
...
-      userWriteProgress = 0;
-    }
+  private long getSyncLagForSpecificConsensusPipe(ConsensusPipeConnector consensusPipeConnector) {
+    long userWriteProgress = consensusPipeConnector.getConsensusPipeCommitProgress();
+    long replicateProgress = consensusPipeConnector.getConsensusPipeReplicateProgress();
+    return Math.max(userWriteProgress - replicateProgress, 0);
   }
 
   public void addConsensusPipeConnector(ConsensusPipeConnector consensusPipeConnector) {
     consensusPipeConnectorList.add(consensusPipeConnector);
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/pipe/metric/PipeConsensusSyncLagManager.java:48]: lock?

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/pipe/metric/PipeConsensusSyncLagManager.java [hunk 2]
```diff
@@ -33,66 +34,50 @@
  * <p>Note: every consensusGroup/dataRegion has and only has 1 instance of this class.
  */
 public class PipeConsensusSyncLagManager {
-  long userWriteProgress = 0;
-  long minReplicateProgress = Long.MAX_VALUE;
+  long syncLag = Long.MIN_VALUE;
+  ReentrantLock lock = new ReentrantLock();
   List<ConsensusPipeConnector> consensusPipeConnectorList = new CopyOnWriteArrayList<>();
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/pipe/metric/PipeConsensusSyncLagManager.java:39]: do we need copy write list anymore if we use lock?

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/pipe/metric/PipeConsensusSyncLagManager.java (hunk 1 above)
Comment (Pengzna) [reply iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/pipe/metric/PipeConsensusSyncLagManager.java:48]: this method is called and is only called in a locked method. so don't need lock here

Comment (Pengzna) [reply iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/pipe/metric/PipeConsensusSyncLagManager.java]: fixed

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/pipe/metric/PipeConsensusSyncLagManager.java (hunk 2 above)
Comment (Pengzna) [reply iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/pipe/metric/PipeConsensusSyncLagManager.java]: fixed

**Pull request merged**